backend/tracim_backend/applications/upload_permissions/controller.py

```diff
@@ -0,0 +1,164 @@
+from tracim_backend.lib.utils.utils import generate_documentation_swagger_tag
+from tracim_backend.views.controllers import Controller
+from tracim_backend.views.core_api.schemas import NoContentSchema
+from tracim_backend.views.core_api.schemas import WorkspaceIdPathSchema
+from tracim_backend.views.core_api.workspace_controller import SWAGGER_TAG__WORKSPACE_ENDPOINTS
+
+SWAGGER_TAG__UPLOAD_PERMISSION_SECTION = "Upload Permission"
+SWAGGER_TAG__WORKSPACE_UPLOAD_PERMISSION_ENDPOINTS = generate_documentation_swagger_tag(
+    SWAGGER_TAG__WORKSPACE_ENDPOINTS, SWAGGER_TAG__UPLOAD_PERMISSION_SECTION
+)
+
+
+def import_controller(
+    configurator: Configurator, app_config: CFG, route_prefix: str
+) -> Configurator:
+    upload_permission_controller = UploadPermissionController()
+    configurator.include(upload_permission_controller.bind, route_prefix=route_prefix)
+    return configurator
+
+
+class UploadPermissionController(Controller):
+    """
+    Endpoints for Upload Permission
+    """
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__WORKSPACE_UPLOAD_PERMISSION_ENDPOINTS])
+    @check_right(is_content_manager)
+    @hapic.input_path(WorkspaceIdPathSchema())
+    @hapic.input_body(UploadPermissionCreationBodySchema())
+    @hapic.output_body(UploadPermissionSchema(many=True))
+    def add_upload_permission(
+        self, context, request: TracimRequest, hapic_data=None
+    ) -> typing.List[UploadPermissionInContext]:
+        """
+        Allow to add permission for upload to external person
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = UploadPermissionLib(
+            current_user=request.current_user, session=request.dbsession, config=app_config
+        )
+        upload_permission = api.add_permission_to_workspace(
+            request.current_workspace, hapic_data.body.emails, hapic_data.body.password
+        )
+        return api.get_upload_permissions_in_context(upload_permission)
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__WORKSPACE_UPLOAD_PERMISSION_ENDPOINTS])
+    @check_right(is_content_manager)
+    @hapic.input_query(UploadPermissionListQuerySchema())
+    @hapic.input_path(WorkspaceIdPathSchema())
+    @hapic.output_body(UploadPermissionSchema(many=True))
+    def get_upload_permission(
+        self, context, request: TracimRequest, hapic_data=None
+    ) -> typing.List[UploadPermissionInContext]:
+        """
+        Get all upload permission of this workspace
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = UploadPermissionLib(
+            current_user=request.current_user,
+            session=request.dbsession,
+            config=app_config,
+            show_disabled=hapic_data.query.show_disabled,
+        )
+        upload_permission = api.get_upload_permissions(request.current_workspace)
+        return api.get_upload_permissions_in_context(upload_permission)
```

Comment:
missing a `s` to `upload_permission ` variable ?

Reply:
done.